lib/lightning/jobs/trigger.ex

```diff
@@ -109,4 +116,20 @@ defmodule Lightning.Jobs.Trigger do
       _ -> changeset
     end
   end
+
+  def with_auth_methods_query do
+    from t in __MODULE__,
+      left_join: wam in assoc(t, :webhook_auth_methods),
+      preload: [webhook_auth_methods: wam],
+      select: %{
+        t
+        | has_auth_method:
+            fragment(
+              "CASE WHEN ? IS NULL THEN ? ELSE ? END",
+              wam.id,
+              false,
+              true
+            )
```

Comment:
I'm not 100% sold on having `has_auth_method` as a virtual field, and also on using a case statement here (I might be wrong here but does it not return duplicates).

Anyway, my main issue with this is, if we're always preloading the auth methods, we can determine if there are any auth methods _afterwards_ with `Enum.any?/1`.